lib/smart_proxy_remote_execution_ssh/session.rb

```diff
@@ -65,6 +76,7 @@ def refresh
     def refresh_command_buffer
       @logger.debug("command #{@command} got new output: #{@command_buffer.inspect}")
       command_update = CommandUpdate.new(@command_buffer)
+      check_expecting_disconnect
```

Comment:
We need to determine, if the disconnection was expected (such as when calling `reboot`) or it was an error. We need to do it on two places, because the `@command_buffer` could be filled in this, or previous iteration.